File: check_tiebreaker_status.py
import psycopg2
import os

conn = psycopg2.connect(os.environ.get('DATABASE_URL') or os.environ.get('NEON_DATABASE_URL'))
cur = conn.cursor()

print("=== TIEBREAKER STATUS ===")
cur.execute("""
    SELECT 
        t.id,
        t.status,
        t.player_id,
        t.winning_team_id,
        t.winning_bid,
        t.round_id,
        r.status as round_status,
        r.position as round_position
    FROM tiebreakers t
    LEFT JOIN rounds r ON t.round_id = r.id
    WHERE t.id = 18
""")
tb = cur.fetchone()

if tb:
    print(f"\nTiebreaker ID: {tb[0]}")
    print(f"Status: {tb[1]}")
    print(f"Player ID: {tb[2]}")
    print(f"Winning Team: {tb[3]}")
    print(f"Winning Bid: {tb[4]}")
    print(f"Round ID: {tb[5]}")
    print(f"Round Status: {tb[6]}")
    print(f"Round Position: {tb[7]}")
else:
    print("Tiebreaker not found")

print("\n=== TEAM TIEBREAKER SUBMISSIONS ===")
cur.execute("""
    SELECT 
        tt.id,
        tt.team_id,
        tt.team_name,
        tt.submitted,
        tt.new_bid_amount
    FROM team_tiebreakers tt
    WHERE tt.tiebreaker_id = 18
""")
submissions = cur.fetchall()

for sub in submissions:
    print(f"\nTeam: {sub[2] or sub[1]}")
    print(f"  Submitted: {sub[3]}")
    print(f"  Bid Amount: {sub[4]}")

print("\n=== ROUND STATUS ===")
if tb:
    cur.execute("""
        SELECT id, position, status, created_at
        FROM rounds
        WHERE id = %s
    """, (str(tb[5]),))
    round_info = cur.fetchone()
    
    if round_info:
        print(f"Round ID: {round_info[0]}")
        print(f"Position: {round_info[1]}")
        print(f"Status: {round_info[2]}")
        print(f"Created: {round_info[3]}")

cur.close()
conn.close()

File: scripts/check-tiebreaker-status.js
const { neon } = require('@neondatabase/serverless');
require('dotenv').config({ path: '.env.local' });

const sql = neon(process.env.DATABASE_URL || process.env.NEON_DATABASE_URL);

const ROUND_ID = process.argv[2] || 'SSPSLFBR00009';

async function checkTiebreakerStatus() {
  console.log(`\n🔍 Checking Tiebreaker Status for Round: ${ROUND_ID}\n`);

  try {
    // Get all tiebreakers for this round
    const tiebreakers = await sql`
      SELECT 
        id,
        player_id,
        player_name,
        status,
        current_highest_team_id,
        current_highest_bid,
        teams_remaining,
        created_at,
        resolved_at
      FROM bulk_tiebreakers
      WHERE bulk_round_id = ${ROUND_ID}
      ORDER BY created_at DESC
    `;

    if (tiebreakers.length === 0) {
      console.log(`⚠️  No tiebreakers found for round ${ROUND_ID}\n`);
      return;
    }

    console.log(`📊 Found ${tiebreakers.length} tiebreaker(s)\n`);
    console.log('='.repeat(80));

    for (const tb of tiebreakers) {
      console.log(`\n🎯 Tiebreaker: ${tb.id}`);
      console.log(`   Player: ${tb.player_name} (ID: ${tb.player_id})`);
      console.log(`   Status: ${tb.status}`);
      console.log(`   Winner Team ID: ${tb.current_highest_team_id || 'None'}`);
      console.log(`   Winning Bid: £${tb.current_highest_bid || 0}`);
      console.log(`   Teams Remaining: ${tb.teams_remaining || 0}`);
      console.log(`   Created: ${new Date(tb.created_at).toLocaleString()}`);
      console.log(`   Resolved: ${tb.resolved_at ? new Date(tb.resolved_at).toLocaleString() : 'Not resolved'}`);

      // Check if player is in team_players
      const teamPlayer = await sql`
        SELECT team_id, season_id, purchase_price, acquired_at
        FROM team_players
        WHERE player_id = ${tb.player_id}
      `;

      if (teamPlayer.length > 0) {
        const tp = teamPlayer[0];
        console.log(`\n   ✅ Player in team_players:`);
        console.log(`      Team: ${tp.team_id}`);
        console.log(`      Season: ${tp.season_id}`);
        console.log(`      Purchase Price: £${tp.purchase_price}`);
        console.log(`      Acquired: ${new Date(tp.acquired_at).toLocaleString()}`);
        
        if (tp.team_id !== tb.current_highest_team_id) {
          console.log(`      ⚠️  WARNING: team_players team (${tp.team_id}) doesn't match tiebreaker winner (${tb.current_highest_team_id})`);
        }
      } else {
        console.log(`\n   ❌ Player NOT in team_players table`);
      }

      // Check teams in this tiebreaker
      const teams = await sql`
        SELECT 
          team_id,
          team_name,
          status,
          current_bid,
          joined_at,
          withdrawn_at
        FROM bulk_tiebreaker_teams
        WHERE tiebreaker_id = ${tb.id}
        ORDER BY current_bid DESC NULLS LAST
      `;

      console.log(`\n   👥 Teams (${teams.length}):`);
      teams.forEach((team, idx) => {
        const isWinner = team.team_id === tb.current_highest_team_id;
        console.log(`      ${idx + 1}. ${team.team_name} (${team.team_id})`);
        console.log(`         Status: ${team.status}${isWinner ? ' 🏆 WINNER' : ''}`);
        console.log(`         Bid: £${team.current_bid || 0}`);
        if (team.withdrawn_at) {
          console.log(`         Withdrawn: ${new Date(team.withdrawn_at).toLocaleString()}`);
        }
      });

      console.log('\n' + '-'.repeat(80));
    }

    console.log('\n📋 Summary:');
    console.log(`   Total tiebreakers: ${tiebreakers.length}`);
    console.log(`   Resolved: ${tiebreakers.filter(t => t.status === 'resolved' || t.status === 'finalized').length}`);
    console.log(`   Active: ${tiebreakers.filter(t => t.status === 'active' || t.status === 'ongoing').length}`);
    console.log(`   Pending: ${tiebreakers.filter(t => t.status === 'pending').length}`);
    
    const withWinner = tiebreakers.filter(t => t.current_highest_team_id).length;
    const withoutWinner = tiebreakers.length - withWinner;
    console.log(`   With winner: ${withWinner}`);
    console.log(`   Without winner: ${withoutWinner}`);

    console.log('\n');

  } catch (error) {
    console.error('❌ Error:', error);
    console.error(error.stack);
    process.exit(1);
  } finally {
    process.exit(0);
  }
}

checkTiebreakerStatus();
